### DISCUSSION TITLE
Problem with internally generated mesh

### DISCUSSION THREAD

**Discussion by joesmerdon** [capytaine/capytaine#399] [Q&A]
I have been using Capytaine to study the interaction of waves with arrays of scatterers.  I am generating the array using the internal cylinder generation routine, generating one cylinder then copying it to all array locations.

My recent simulations have failed, citing a NaN returned during one of the solver operations.  The error thrown says that the problem might be overlapping panels.  I have checked and there appear to be no overlaps between cylinders or duplicate array locations.

Any ideas?

Thanks

Joe

**Comment by mancellin** [comment]
That is not much information to figure out what is going on.
If not overlapping panels, it could be panels on or above the free surface, or below the sea bottom (but it should be caught by another error message).
Could you try to extract a small self-contained Python script that exhibits the issue for me to investigate?

**Comment by joesmerdon** [comment]
Thanks for the response.  Sorry that the information was insufficient.  I tracked the issue down myself.

FYI:
My calculations take in the simulation description in a .csv file, in which the first line is 
`x-size,y-size,,`
of the simulation (the trailing commas are to make space for more columns further down).  The first value was being read as a NaN, so it could build the structure mesh fine, but crashed when asked to compute the free surface mesh. This is reproducible, but doesn't happen with all input files.

I still don't know why the first value is NaN.  I got around it by adding a 0 before that, so now the first line is 
`0,x-size,y-size,`

Edit:

I opened a working and non-working (without workaround) file in HxD to see if there was anything I couldn't see in Notepad++.
There, of course, is!

![image](https://github.com/capytaine/capytaine/assets/44434608/05932fe8-840d-4c30-bcb0-28bead1335e2)

The few characters before the 0 are only present in non-working files.

**Reply by mancellin** [reply]
Thank for the explanation.
Is there something that could be changed in Capytaine to help? Maybe checking for NaNs at pre-processing and not only at post-processing?
